scm-ui/ui-webapp/src/containers/Notifications.tsx

```diff
@@ -0,0 +1,337 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020-present Cloudogu GmbH and Contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import React, { FC, useEffect, useState } from "react";
+import {
+  Button,
+  Notification as InfoNotification,
+  ErrorNotification,
+  Icon,
+  ToastArea,
+  ToastNotification,
+  ToastType,
+  Loading,
+  DateFromNow,
+  devices
+} from "@scm-manager/ui-components";
+import styled from "styled-components";
+import {
+  useClearNotifications,
+  useDismissNotification,
+  useNotifications,
+  useNotificationSubscription
+} from "@scm-manager/ui-api";
+import { Notification, NotificationCollection } from "@scm-manager/ui-types";
+import { useHistory, Link } from "react-router-dom";
+import classNames from "classnames";
+import { useTranslation } from "react-i18next";
+
+const Bell = styled(Icon)`
+  font-size: 1.5rem;
+`;
+
+const Container = styled.div`
+  display: flex;
+  cursor: pointer;
+
+  @media screen and (max-width: ${devices.desktop.width}px) {
+    padding-right: 1rem;
+  }
+`;
+
+const DropDownMenu = styled.div`
+  min-width: 35rem;
+
+  @media screen and (max-width: ${devices.mobile.width}px) {
+    min-width: 25rem;
+  }
+
+  @media screen and (max-width: ${devices.desktop.width}px) {
+    margin-right: 1rem;
+  }
+
+  &:before {
+    position: absolute;
+    content: "";
```

Comment:
This is a small css hack which allows us to style dynamic content, because it is not possible to apply styles to generated content without defining the content. The whole "&:before" generates the small triangle on the notification box which points to the bell icon.